src/services/BrowserPasswordReprompt.service.ts

```diff
@@ -0,0 +1,40 @@
+import Swal from 'sweetalert2';
+
+import { CryptoService } from '../abstractions/crypto.service';
+import { I18nService } from '../abstractions/i18n.service';
+import { PasswordRepromptService } from '../abstractions/passwordReprompt.service';
+
+export class BrowserPasswordRepromptService implements PasswordRepromptService {
+    constructor(private i18nService: I18nService, private cryptoService: CryptoService) { }
+
+    protectedFields() {
+        return ['TOTP', 'Password', 'H_Field', 'Card Number', 'Security Code'];
+    }
+
+    async showPasswordPrompt() {
+        const result = await Swal.fire({
```

Comment:
This seems weird to me to be putting in jslib services. I would expect Swal to be a client-specific implementation of a UI prompt. Perhaps abstracted out into a platform utils function that is called directly here or with some kind of messaging.

Reply:
Yea, I initially wanted to put it in the `PlatformUtilsService` but since it requires the `cryptoService` that creates a cyclic dependency. Hence I extracted it to a separate service which was supposed to be client specific, but it turned out that each of the jsclient could share the same implementation using Swal.

Using messages to communicate with `platformUtilsService` would be nice, but might be tricky using the `inputValidator`.

Reply:
We could refactor it into:

```typescript
async showPasswordPrompt() {
  const passwordValidation = (value: string): Promise<boolean> => {
    return password == expectedhash
  }
  
  this.platformUtilsService.passwordDialog(title, text, validator);
}
```

Reply:
@kspearrin I refactored it, had to resort to messaging in the browser extension https://github.com/bitwarden/browser/pull/1784/files#diff-027b4bec7e9f506ae15598d89d8a2f7754776435e2545a8e868d61f40a3de158 but all in all it seems to work fine.